File: src/main/java/org/samoxive/safetyjim/server/entities/GuildEntity.java
package org.samoxive.safetyjim.server.entities;

import java.util.List;

public class GuildEntity {
    public String id;
    public String name;
    public String avatarUrl;
    public List<PartialChannel> channels;

    public GuildEntity(String id, String name, String avatarUrl, List<PartialChannel> channels) {
        this.id = id;
        this.name = name;
        this.avatarUrl = avatarUrl;
        this.channels = channels;
    }
}

File: src/main/java/org/samoxive/safetyjim/server/entities/GuildEntity.kt
package org.samoxive.safetyjim.server.entities

import kotlinx.serialization.Serializable
import net.dv8tion.jda.api.entities.Guild
import org.samoxive.safetyjim.tryhard

@Serializable
data class GuildEntity(
    val id: String,
    val name: String,
    val iconUrl: String
)

fun Guild.toGuildEntity(): GuildEntity = GuildEntity(id, name, tryhard { iconUrl } ?: "")